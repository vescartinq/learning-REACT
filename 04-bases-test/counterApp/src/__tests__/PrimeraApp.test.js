import React from 'react';
// import { render } from '@testing-library/react';
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';

import PrimeraApp from '../PrimeraApp';

describe.skip('PrimeraApp', () => {
  // ------------JEST-----------
  // test('should show message "Hola, soy Goku', () => {
  //   const greeting = 'Hola, soy Goku';

  //   const wrapper = render(<PrimeraApp saludo={greeting} />);

  //   expect(wrapper.getByText(greeting)).toBeInTheDocument();
  // });

  // -----------ENZYME----------
  test('should show PrimeraApp component correctly', () => {
    const greeting = 'Hola, soy Goku';
    const wrapper = shallow(<PrimeraApp saludo={greeting} />);

    expect(wrapper).toMatchSnapshot();
  });

  test('should show the subtitle sent by props', () => {
    const greeting = 'Hola, soy Goku';
    const subtitle = 'Soy el subtitulo';
    const wrapper = shallow(
      <PrimeraApp saludo={greeting} subtitulo={subtitle} />
    );

    const textoParrafo = wrapper.find('p').text();
    // console.log(textoParrafo);

    expect(textoParrafo).toBe(subtitle);
  });
});
