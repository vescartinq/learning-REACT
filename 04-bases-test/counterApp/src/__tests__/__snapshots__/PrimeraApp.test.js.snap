// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PrimeraApp should show PrimeraApp component correctly 1`] = `
<Fragment>
  <h1>
     
    Hola, soy Goku
     
  </h1>
  <p>
    Soy un subtitulo
  </p>
</Fragment>
`;
